### ISSUE TITLE
REF operator

### ISSUE THREAD

**Issue by Lurgrid** [kubo/rust-oracle#92]
Hello, 

When I run a query that has the operator ``REF()`` I get an error ``Error: unknown Oracle type number 0``. 

__Exemple__ 
```sql
SELECT REF(X) 
FROM INTERVENANT X 
WHERE X.MAT = 1000;
```

Here is the doc for ``REF()``

https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/REF.html

**Comment by Lurgrid** [comment]
I've just noticed that if a table contains an element of type ``REF TYPE`` then I can't display it. 

```sql
CREATE TYPE MISSION_TYPE AS OBJECT (
    CODE VARCHAR2(64),
    INTITULE VARCHAR2(64),
    NB_JOURS INTEGER,
    INTERVENANT REF INTERVENANT_TYPE
);

CREATE TABLE MISSION
OF MISSION_TYPE (
    CONSTRAINT PK_CODE PRIMARY KEY (CODE)
);

SELECT * FROM MISSION
```

**Comment by cjbj** [comment]
What is `INTERVENANT_TYPE` defined as?

**Comment by kubo** [comment]
[`REF` data type](https://docs.oracle.com/en/database/oracle/oracle-database/23/lnoci/data-types.html#GUID-43D2E2C0-E0B1-4712-AAEC-17DC8B84AE7B) isn't supported by rust-oracle. It is an ODPI-C limitation as @cjbj mentioned.

Oracle Call Interface (OCI) supports it. Its data type code is [`SQLT_REF`](https://docs.oracle.com/en/database/oracle/oracle-database/23/lnoci/data-types.html#GUID-43D2E2C0-E0B1-4712-AAEC-17DC8B84AE7B). [ODPI-C](https://oracle.github.io/odpi/) converts the OCI data type code to [Oracle type number](https://odpi-c.readthedocs.io/en/latest/enums/dpiOracleTypeNum.html) in [`dpiOracleType__convertFromOracle`](https://github.com/oracle/odpi/blob/v5.4.0/src/dpiOracleType.c#L375-L452). However `SQLT_REF` isn't handled in the function and it returns `(dpiOracleTypeNum)0`. So the error is `Error: unknown Oracle type number 0`.

**Comment by cjbj** [comment]
@Lurgrid were you just testing REF, or do you have a business need for support?

**Comment by Lurgrid** [comment]
Yes it happened during a normal use of my program. But if that's not possible, I can at worst use the ``sqlplus`` binary directly and convert its responses.

**Comment by cjbj** [comment]
I can't predict when our ODPI-C or @kubo's rust-oracle will support the feature, but I have noted your vote for it.
